pkg/envoy/accesslog.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2017, 2018 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package envoy
+
+import (
+	"github.com/golang/protobuf/proto"
+	"net"
+	"os"
+	"path/filepath"
+	"strings"
+	"syscall"
+)
+
+func getAccessLogPath(stateDir string) string {
+	return filepath.Join(stateDir, "access_log.sock")
+}
+
+// StartAccessLogServer starts the access log server.
+func StartAccessLogServer(stateDir string) {
+	accessLogPath := getAccessLogPath(stateDir)
+	StartAccessLogServer(accessLogPath)
+
+	// Create the access log listener
+	os.Remove(accessLogPath) // Remove/Unlink the old unix domain socket, if any.
+	accessLogListener, err := net.ListenUnix("unixpacket", &net.UnixAddr{Name: accessLogPath, Net: "unixpacket"})
+	if err != nil {
+		log.WithError(err).Fatal("Envoy: Failed to listen at ", accessLogPath)
```

Comment:
There's a missing `%s` and should be `Fatalf`

Reply:
Fixed.

Reply:
Did a pass of cleanup on all log messages.